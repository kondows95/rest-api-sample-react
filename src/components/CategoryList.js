import React from 'react'
import PropTypes from 'prop-types'
import { makeStyles } from '@material-ui/core/styles'
import { 
  Box, Container, Paper, Grid, Button,
  Dialog, DialogTitle, DialogContent, DialogActions, TextField, 
} from '@material-ui/core'
import { validateForm } from '../util'

const useStyles = makeStyles(theme => ({
  button: {
    marginRight: 0,
    marginLeft: "auto",
  },
  paper: {
    display: 'flex',
    margin: theme.spacing(2, 1),
    padding: theme.spacing(2, 2),
  },
}))

const CategoryList = ({ categories, saveCategory, deleteCategory }) => {
  const classes = useStyles()
  const [dialogOpen, setDialogOpen] = React.useState(false)
  const [selectedCategory, setSelectedCategory] = React.useState(null)
  const [isDelete, setIsDelete] = React.useState(false)
  const [errors, setErrors] = React.useState({})
  
  const initialCategory = {id: null, name: ""}
  
  const handleChangeValue = fieldName => event => {
    const newCategory = {...selectedCategory}
    newCategory[fieldName] =  event.target.value
    setSelectedCategory(newCategory)
  }
  
  const handleDelete = category => event => {
    setSelectedCategory(category)
    setDialogOpen(true)
    setIsDelete(true)
    setErrors({})
  }
  
  const handleEdit = category => event => {
    setSelectedCategory(category)
    setDialogOpen(true)
    setIsDelete(false)
    setErrors({})
  }

  const handleCloseDialog = () => {
    setDialogOpen(false)
    setSelectedCategory(null)
  }
  
  const validationSetting = {
    isEmpty: ['name'],
  }
  
  const handleSubmit = () => {
    if (selectedCategory) {
      const errs = validateForm(validationSetting, selectedCategory)
      if (errs) {
        setErrors(errs)
      }
      else {
        if (isDelete) {
          deleteCategory(selectedCategory.id)
        }
        else {
          saveCategory(selectedCategory)
        }
        handleCloseDialog()
      }
    }
  }
  
  const deleteDialog = (selectedCategory && isDelete) ? (
    <Dialog 
      open={dialogOpen} 
      onClose={handleCloseDialog} 
      aria-labelledby="category-delete-dialog"
      fullWidth
      maxWidth="xs"
    >
      <DialogTitle id="category-delete-dialog">
        {"Are you sure?"}
      </DialogTitle>
      <DialogContent>
        <Box>Do you really want to delete {selectedCategory.id}: {selectedCategory.name}.</Box>
        <Box fontWeight={600}></Box>
      </DialogContent>
      <DialogActions>
        <Button onClick={handleCloseDialog} color="primary">
          Cancel
        </Button>
        <Button onClick={handleSubmit} color="primary">
          Delete
        </Button>
      </DialogActions>
    </Dialog>
  ) : null
  
  const saveDialog = (selectedCategory && isDelete === false) ? (
    <Dialog 
      open={dialogOpen} 
      onClose={handleCloseDialog} 
      aria-labelledby="category-save-dialog"
      fullWidth
      maxWidth="xs"
    >
      <DialogTitle id="category-save-dialog" >
        {selectedCategory.id ? "Edit (ID:"+selectedCategory.id+")" : "Create"}
      </DialogTitle>
      <DialogContent>
        <TextField
          variant="outlined"
          autoFocus
          error={errors.name ? true : false}
          id="name"
          label="Name"
          value={selectedCategory.name}
          onChange={handleChangeValue("name")}
          fullWidth
        />
      </DialogContent>
      <DialogActions>
        <Button id = "cancle" onClick={handleCloseDialog} color="primary">
          Cancel
        </Button>
        <Button id = "submit" onClick={handleSubmit} color="primary">
          Submit
        </Button>
      </DialogActions>
    </Dialog>
  ) : null
  
  const paperItems = []
  for (const category of categories) {
    paperItems.push(
      <Grid key={category.id} item xs={12} sm={4} lg={3}>
        <Paper className={classes.paper}>
          <Box display="flex" flexDirection="column" flexGrow={1}>
            <Box fontWeight={600}>{category.id}: {category.name}</Box>
            <Box ml="auto" mr={0} mt={2}>
              <Button color="primary" onClick={handleEdit(category)}>
                Edit
              </Button>
              <Button color="primary" onClick={handleDelete(category)}>
                Delete
              </Button>
            </Box>
          </Box>
        </Paper>
      </Grid>
    )
  }
  
  const paperControl = (
    <Paper className={classes.paper}>
      <Box ml={0} my="auto" fontWeight={600}>
        Categories ({categories.length})
      </Box>
      <Button 
        id="create"
        variant="contained" 
        color="secondary" 
        className={classes.button}
        onClick={handleEdit(initialCategory)}
      >
        Create
      </Button>
    </Paper>
  )
  
  return (
    <React.Fragment>
      <Container maxWidth="lg">
        {paperControl}
        <Grid container>
          {paperItems}
        </Grid>
      </Container>
      {saveDialog}
      {deleteDialog}
    </React.Fragment>
  )
}

const categoryPropTypes = PropTypes.shape({
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
})

CategoryList.propTypes = {
  categories: PropTypes.arrayOf(categoryPropTypes.isRequired).isRequired,
}

export default CategoryList
