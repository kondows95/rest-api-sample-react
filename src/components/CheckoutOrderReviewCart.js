import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { BASEURL_ITEM_IMAGES } from '../constants'
import { Grid, Box, CardMedia, Paper, Divider, Hidden } from '@material-ui/core';

const useStyles = makeStyles(theme => ({
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
  descImagePC: {
    width: 70,
    height: 100
  },
  menu: {
    width: 200,
  },
}));

const CheckoutOrderReviewCart = ({ cart, totalPrice }) => {
  const classes = useStyles()
  const mobileView = [];
  const pcView = [];
  const Netprice = () => (
    <Box display="flex" mt={2} pl={3} pb={3}>
      <Grid item xs={false} sm={4}>
        <Box> </Box>
      </Grid>
      <Grid item xs={6} sm={4}>
        <Box> SUBTOTAL </Box>
        <Box> SHIPPING COSTS </Box>
        <Box> ADDITIONAL SHIPPING COSTS</Box>
        <Box>SALES TAX</Box>
        <Box>TOTAL COSTS(4 ARTICLSE)</Box>
      </Grid>
      <Grid item xs={6} sm={4} >
        <Box textAlign="right" pr={3}> {totalPrice} MMK </Box>
        <Box textAlign="right" pr={3}> 0.0 MMK </Box>
        <Box textAlign="right" pr={3}> 0.0 MMK </Box>
        <Box textAlign="right" pr={3}> 0.0 MMK </Box>
        <Box textAlign="right" pr={3}> {totalPrice} MMK</Box>
      </Grid>
    </Box>
  )
  cart.map((cartList, k) => {
    pcView.push(
      <Grid container key={cartList.id}>
        <Box flexGrow={1} display="flex">
          <Grid item xs={3}>
            <Box textAlign="left" my={2} pl={3}>
              <CardMedia
                className={classes.descImagePC}
                image={BASEURL_ITEM_IMAGES + cartList.image}
                title={cartList.name}
              />
            </Box>
          </Grid>
          <Grid item xs={3}>
            <Box textAlign="left" my={2} fontWeight="100">
              {cartList.name}
            </Box>
          </Grid>
          <Grid item xs={2}>
            <Box textAlign="left" my={2} >
              {cartList.quantity}x
                        </Box>
          </Grid>
        </Box>
        <Grid item xs={2}>
          <Box textAlign="right" my={2} pr={3}>
            <Box pb={2}>{cartList.quantity}x{cartList.price} MMK</Box>
            <Box> {cartList.subTotal} MMK </Box>
          </Box>
        </Grid>
        <Grid item xs={12}><Divider /></Grid>
      </Grid>
    )
    
    return null;
  });
  
  cart.map((cartList, k) => {
    mobileView.push(
      <Grid container key={cartList.id}>
        <Box flexGrow={1} display="flex">
          <Grid item xs={6}>
            <Box textAlign="left" my={2} pl={3}>
              <CardMedia
                className={classes.descImagePC}
                image={BASEURL_ITEM_IMAGES + cartList.image}
                title={cartList.name}
              />
            </Box>
          </Grid>
          <Grid item xs={6}>
            <Box textAlign="right" my={2} pr={3} fontWeight="fontWeightBold">
              {cartList.name}
            </Box>
            <Box textAlign="right" my={2} pr={3} >
              {cartList.quantity}x {cartList.price} MMK
                        </Box>
            <Box textAlign="right" my={2} pr={3}>
              {cartList.subTotal} MMK
                        </Box>
          </Grid>
        </Box>
        <Grid item xs={12}><Divider /></Grid>
      </Grid>
    )
    
    return null;
  })
  return (
    <Paper>
      <Hidden xsDown>
        <Grid key={0} container>
          <Box flexGrow={1} display="flex" >
            <Grid item xs={3} >
              <Box textAlign="left" my={2} pl={3}>PRODUCT</Box>
            </Grid>
            <Grid item xs={3}>
              <Box textAlign="left" my={2} xs={3} >DESCRIPTION</Box>
            </Grid>
            <Grid item xs={2}>
              <Box textAlign="left" my={2} xs={3}>QUANTITY</Box>
            </Grid>
          </Box>
          <Grid item xs={2}>
            <Box textAlign="right" my={2} pr={3} xs={3}>PRICE</Box>
          </Grid>
          <Grid item xs={12}><Divider /></Grid>
        </Grid>
        {pcView}
      </Hidden>
      <Hidden smUp>
        {mobileView}
      </Hidden>
      <Netprice />
    </Paper>
  );
}
export default CheckoutOrderReviewCart;
