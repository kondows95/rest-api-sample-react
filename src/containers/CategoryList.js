import { connect } from 'react-redux'
import CategoryList from '../components/CategoryList'
import { saveCategory, deleteCategory, dialogBox } from '../modules/categories'

export default connect(
  (state) => ({
    items: state.items.rows,
    categories: state.categories.rows,
    loading:state.categories.loading,
    closeDialog:state.categories.closeDialog
  }),
  (dispatch) => ({
    saveCategory: (category) =>  dispatch(saveCategory(category)),
    deleteCategory: (id) =>  dispatch(deleteCategory(id)),
    dialogBox:(value) => dispatch(dialogBox(value)),
  })
)(CategoryList)
