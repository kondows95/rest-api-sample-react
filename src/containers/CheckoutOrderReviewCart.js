import { connect } from 'react-redux'
import CheckoutOrderReviewCart from '../components/CheckoutOrderReviewCart'

export default connect(
  (state) => ({
    cart: state.cart.rows, totalPrice: state.cart.totalPrice
  }),
  null
)(CheckoutOrderReviewCart)