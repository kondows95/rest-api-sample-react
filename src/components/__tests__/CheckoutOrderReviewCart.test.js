import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import CategoryList from '../CategoryList';
import renderer from 'react-test-renderer';
import CheckoutOrderReviewCart from '../CheckoutOrderReviewCart';
let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.outerHTML = '<body></body>'
  container = null
})

const valueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "value").set;

describe("CheckoutOrderReviewCart component", () => {
  let cart = [{ id: 1, name: "a", quantity: 1, price: 1000, image:"a.jpg",subTotal : 1000 },
  { id: 2, name: "b", quantity: 1, price: 1000 , image:"a.jpg",subTotal : 1000 }];

  it('matches the snapshot', () => {
    const OrderReviewCartSnapshot = renderer.create(<CheckoutOrderReviewCart
      cart={cart}
      totalPrice={2000}
    />).toJSON();
    expect(OrderReviewCartSnapshot).toMatchSnapshot();
  });
});

