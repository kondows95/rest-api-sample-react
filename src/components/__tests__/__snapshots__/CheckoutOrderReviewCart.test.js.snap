// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutOrderReviewCart component matches the snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiBox-root MuiBox-root-33"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4"
    >
      <div
        className="MuiBox-root MuiBox-root-137"
      >
         
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4"
    >
      <div
        className="MuiBox-root MuiBox-root-138"
      >
         SUBTOTAL 
      </div>
      <div
        className="MuiBox-root MuiBox-root-139"
      >
         SHIPPING COSTS 
      </div>
      <div
        className="MuiBox-root MuiBox-root-140"
      >
         ADDITIONAL SHIPPING COSTS
      </div>
      <div
        className="MuiBox-root MuiBox-root-141"
      >
        SALES TAX
      </div>
      <div
        className="MuiBox-root MuiBox-root-142"
      >
        TOTAL COSTS(4 ARTICLSE)
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4"
    >
      <div
        className="MuiBox-root MuiBox-root-143"
      >
         
        2000
         MMK 
      </div>
      <div
        className="MuiBox-root MuiBox-root-144"
      >
         0.0 MMK 
      </div>
      <div
        className="MuiBox-root MuiBox-root-145"
      >
         0.0 MMK 
      </div>
      <div
        className="MuiBox-root MuiBox-root-146"
      >
         0.0 MMK 
      </div>
      <div
        className="MuiBox-root MuiBox-root-147"
      >
         
        2000
         MMK
      </div>
    </div>
  </div>
</div>
`;
